influxdb3_catalog/src/catalog.rs

```diff
@@ -388,42 +386,62 @@ impl InnerCatalog {
         let table_count = self.table_count();
 
         if let Some(db) = self.databases.get(&catalog_batch.database_id) {
-            let existing_table_count = db.tables.len();
-
-            if let Some(new_db) = db.new_if_updated_from_batch(catalog_batch)? {
-                let new_table_count = new_db.tables.len() - existing_table_count;
-                if table_count + new_table_count > Catalog::NUM_TABLES_LIMIT {
-                    return Err(Error::TooManyTables);
-                }
-                let new_db = Arc::new(new_db);
-                self.databases.insert(new_db.id, Arc::clone(&new_db));
-                self.sequence = self.sequence.next();
-                self.updated = true;
-                self.db_map.insert(new_db.id, Arc::clone(&new_db.name));
+            if let Some(new_db) = DatabaseSchema::new_if_updated_from_batch(db, catalog_batch)? {
+                check_overall_table_count(db, &new_db, table_count)?;
+                self.upsert_db(new_db);
             }
         } else {
-            if self.databases.len() >= Catalog::NUM_DBS_LIMIT {
-                return Err(Error::TooManyDbs);
-            }
-
-            let new_db = DatabaseSchema::new_from_batch(catalog_batch)?;
-            if table_count + new_db.tables.len() > Catalog::NUM_TABLES_LIMIT {
-                return Err(Error::TooManyTables);
-            }
-
-            let new_db = Arc::new(new_db);
-            self.databases.insert(new_db.id, Arc::clone(&new_db));
-            self.sequence = self.sequence.next();
-            self.updated = true;
-            self.db_map.insert(new_db.id, Arc::clone(&new_db.name));
+            let new_db = self.check_db_count(catalog_batch, table_count)?;
+            self.upsert_db(new_db);
```

Comment:
IMHO the `upsert_db` method is nice here to DRY things up, but the logic captured in the `check_overall_table_count` and `Self::check_db_count` is simple enough to keep in line. Furthermore, I don't think their names as is capture what they are doing very well.

Reply:
Sure, think `check_db_count` does a bit more creates db and checks table count again. I'll rejig it and see if you still think it should be in-lined. Having said that am I missing something in `check_overall_table_count`? I don't mind in-lining by the way, just wanted to see if I've missed anything.

Reply:
No I don't think you missed anything the logic is still good. I'm being pedantic, but felt that moving the logic out to a helpers that aren't re-used otherwise was overkill.

Reply:
Yea definitely, what I wanted to tease out was the table count check as that gets used in both branches. I've kept the db check and creation inline now.